tests/functional/ComboPopup.js

```diff
@@ -403,7 +403,127 @@ define([
 						valueNodeValueAtLatestInputEvent: "China,Germany",
 						widgetValueAtLatestChangeEvent: ["China", "Germany"],
 						valueNodeValueAtLatestChangeEvent: "China,Germany"
-					}, "after clicking again the root node (close)");
+					}, "after clicking on the 'Ok' button (close)");
+			})
+			.end();
+	};
+
+	var checkMultiSelectionCancel = function (remote, comboId) {
+		var executeExpr = "return getComboState(\"" + comboId + "\");";
+		return loadFile(remote, "./ComboPopup.html")
+			.findById(comboId)
+			.execute(executeExpr)
+			.then(function (comboState) {
+				// The first option should be the one selected,
+				// the popup is closed initially
+				checkComboState(comboId, comboState,
+					{ // expected combo state
+						inputNodeValue: comboState.multipleChoiceNoSelectionMsg,
+						popupInputNodeValue: "", // invisible!
+						widgetValue: [],
+						valueNodeValue: "",
+						opened: false,
+						selectedItemsCount: 0,
+						itemRenderersCount: 37,
+						inputEventCounter: 0, // no event so far
+						changeEventCounter: 0,
+						widgetValueAtLatestInputEvent: undefined, // never received
+						valueNodeValueAtLatestInputEvent: undefined,
+						widgetValueAtLatestChangeEvent: undefined,
+						valueNodeValueAtLatestChangeEvent: undefined
+					}, "right after load");
+			})
+			.click()
+			.sleep(500) // wait for List's loading panel to go away
+			.execute(executeExpr)
+			.then(function (comboState) {
+				// The first option should be the one selected,
+				// the first click just opens the dropdown.
+				checkComboState(comboId, comboState,
+					{ // expected combo state
+						inputNodeValue: comboState.multipleChoiceNoSelectionMsg,
+						popupInputNodeValue: "", // invisible
+						widgetValue: [],
+						valueNodeValue: "",
+						opened: true,
+						selectedItemsCount: 0,
+						itemRenderersCount: 37,
+						inputEventCounter: 0, // no event so far
+						changeEventCounter: 0,
+						widgetValueAtLatestInputEvent: undefined, // never received
+						valueNodeValueAtLatestInputEvent: undefined,
+						widgetValueAtLatestChangeEvent: undefined,
+						valueNodeValueAtLatestChangeEvent: undefined
+					}, "after click on root node");
+			})
+			.end()
+			.findById(comboId + "_item1") // "Germany"
+			.click()
+			.sleep(500) // wait for popup to close
```

Comment:
Actually, now I see that `checkMultiSelection()` and `checkMultiSelectionCancel()` are basically the same code, except for the last section where `checkMultiSelection()` clicks OK and `checkMultiSelectionCancel()` clicks Cancel.    So instead of having all that copy-and-pasted code can you just make a single method?

You could make `checkMultiSelection()` take a parameter (or parameters) to control whether it clicks the OK button or the cancel button.   Or alternately you could move the code that's specific to each test out of the shared function and into the tests themselves.